src/main/java/uk/gov/companieshouse/accountsDates/AccountsDates.java

```diff
@@ -0,0 +1,199 @@
+	 * 	Generate balance sheet header string based on 
+	 *  period start and end dates
+	 * 
+	 */
+	public String generateBalanceSheetHeading(String periodStartString, String periodEndString, boolean isSameYear) {
+
+		Instant instantPeriodStart = Instant.parse(periodStartString);
+		Instant instantPeriodEnd = Instant.parse(periodEndString);
+
+		LocalDateTime localDateTimePeriodStart = LocalDateTime.ofInstant(instantPeriodStart, ZoneId.of("GMT"));
+		LocalDate localDatePeriodStart = localDateTimePeriodStart.toLocalDate();
+
+		LocalDateTime localDateTimePeriodEnd = LocalDateTime.ofInstant(instantPeriodEnd, ZoneId.of("GMT"));
+		LocalDate localDatePeriodEnd = localDateTimePeriodEnd.toLocalDate();
+
+		Map<String, String> resultdates = calculatePeriodRange(localDatePeriodStart, localDatePeriodEnd, isSameYear);
+
+		if (resultdates.get("periodStart") == null) {
+			return resultdates.get("periodEnd");
+		} else {
+			return resultdates.get("periodStart") + " to " + resultdates.get("periodEnd");
+		}
+	}
+
+	/**
+	 * @param periodStart
+	 * @param periodEnd
+	 * @param isSameYear
+	 * @return
+	 * 
+	 * 	Calculate balance sheet dates display format depending on range between
+	 *  period start and end dates
+	 * 
+	 */
+	public Map<String, String> calculatePeriodRange(LocalDate periodStart, LocalDate periodEnd, boolean isSameYear) {
+
+		int allowance = 15; // We allow year +/-15 days difference between dates to treat them as full year difference
+		int yearRangeMax = 365 + allowance;
+		int yearRangeMin = 365 - allowance;
+
+		Map<String, String> periodObject = new HashMap<>();
+
+		// period returns years, hours and days between dates eg 1 year, 3 months and 4 days
+		Period accountsPeriod = Period.between(periodStart, periodEnd);
+
+		long totalDaysDiff = ChronoUnit.DAYS.between(periodStart, periodEnd) + 1; // + 1 as they have time till the end of the day
+		long totalMonthsDiff = ChronoUnit.MONTHS.between(periodStart, periodEnd);
+
+		// If the days over an exact month are 1 to 15, then round down (unless thegperiod is less than 1 month, in which case round up).
+		// If the days over are 16 to 31, then round up. e.g. 15 months and 4 days is
+		// shown as 15 months; 15 months and 21 days is shown as 16 months.
+		totalMonthsDiff = (accountsPeriod.getDays() >= 15) ? totalMonthsDiff + 1 : totalMonthsDiff; // + 1 to days to include end date in calculation
+
+		// If the previous and current periods both end in the same year, then the
+		// heading is output as a full date e.g. ‘5 January 2015’ ’31 December 2015’.
+		if (isSameYear) {
+			periodObject.put(PERIOD_START, null);
+			periodObject.put(PERIOD_END, convertStringToDisplayDate(convertDateToString(periodEnd)));
+		}
+
+		// If the accounting period is twelve months (+/- 15 days - period start to
+		// period end)
+		// Then the heading for the balance sheet figures is: ccyy e.g. 2015 2016
+		else if ((totalDaysDiff >= yearRangeMin && totalDaysDiff <= yearRangeMax) && totalMonthsDiff == 12) {
```

Comment:
Instead of using else if statements can you return the period object to make this easier to read please?

e.g.
```java
if (isSameYear) {
   // Set period end
   return periodObject
}

if(periodGreaterThanTwelveMonths()) {
   // set data in periodObject
   return periodObject
}

if(totalDaysDiff < yearRangeMin || totalDaysDiff > yearRangeMax) {
   // set data in periodObject
   return periodObject
}

reutrn periodObject
```